lib/galaxy/datatypes/binary.py

```diff
@@ -1224,6 +1224,56 @@ def display_peek(self, dataset):
             return f"Binary Anndata file ({nice_size(dataset.get_size())})"
 
 
+class Grib(Binary):
+    """
+    Class describing an GRIB file
+
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> fname = get_test_fname('test.grib')
+    >>> Grib().sniff(fname)
+    True
+    >>> fname = get_test_fname('interval.interval')
+    >>> Grib().sniff(fname)
+    False
+    """
+    file_ext = "grib"
+    # GRIB not yet in EDAM (work in progress). For now, so set to binary
+    edam_format = "format_2333"
+
+    def __init__(self, **kwd):
+        super().__init__(**kwd)
+        self._magic = b'GRIB'
+
+    def sniff(self, filename):
+        # The first 4 bytes of any GRIB file are GRIB
+        try:
+            _uint8struct = struct.Struct(b">B")
+            f = open(filename, 'rb')
+            header = f.read(4)
+            if header == self._magic:
+                tmp = f.read(4)
+                edition = _uint8struct.unpack_from(tmp, 3)[0]
+                if edition == 1 or edition == 2:
+                    return True
```

Comment:
Is that edition important for you? If so you could save it as metadata.

Reply:
I added it because I was a bit worried to only check 'GRIB' but it is a good idea to save it as metadata. I am not sure how to do it properly. 
1. Add MetadataElement: 
` MetadataElement(name="grib_edition", default=1, desc="GRIB edition", readonly=True, visible=True, optional=True, no_value=0)`
2. Add `set_meta`:
```
def set_meta(self, dataset, **kwd):
        """
        Set the GRIB edition.
        """
        dataset.metadata.grib_edition = self._get_grib_edition(dataset.file_name)

def _get_grib_edition(self, filename):
         _uint8struct = struct.Struct(b">B")
         edition = 0
          with open(filename) as f:
               header = f.read(4)
               tmp = f.read(4)
               edition = _uint8struct.unpack_from(tmp, 3)[0]
           return edition
```

Would it work? It is sufficient or is there anything else to add? Thanks!

Reply:
that looks ok to me